src/envoy/http/mixer/filter.cc

```diff
@@ -192,6 +237,38 @@ void Filter::completeCheck(const Status& status) {
   }
 
   state_ = Complete;
+  route_directive_ = info.route_directive;
+
+  // handle direct response from the route directive
+  if (status.ok() && route_directive_.direct_response_code() != 0) {
+    ENVOY_LOG(debug, "Mixer::Filter direct response");
+    state_ = Responded;
+    decoder_callbacks_->sendLocalReply(
+        Code(route_directive_.direct_response_code()),
+        route_directive_.direct_response_body(), nullptr);
+    return;
+  }
+
+  // handle request header operations
+  for (auto const iter : route_directive_.update_request_headers()) {
```

Comment:
can we share a function for this one with update_response_headers?

Reply:
Done